packages/swapper/src/swappers/JupiterSwapper/swapperApi/getTradeQuote.ts

```diff
@@ -0,0 +1,231 @@
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!isSupportedChainId(buyAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  const maybeQuoteResponse = await getJupiterQuote({
+    apiUrl: jupiterUrl,
+    sourceAsset: sellAsset.assetId === solAssetId ? wrappedSolAssetId : sellAsset.assetId,
+    destinationAsset: buyAsset.assetId === solAssetId ? wrappedSolAssetId : buyAsset.assetId,
+    commissionBps: affiliateBps,
+    amount: sellAmount,
+    slippageBps: convertDecimalPercentageToBasisPoints(
+      slippageTolerancePercentageDecimal ??
+        getDefaultSlippageDecimalPercentageForSwapper(SwapperName.Jupiter),
+    ).toFixed(),
+  })
+
+  if (maybeQuoteResponse.isErr()) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'Quote request failed',
+        code: TradeQuoteError.NoRouteFound,
+      }),
+    )
+  }
+
+  const { data: quoteResponse } = maybeQuoteResponse.unwrap()
+
+  const getFeeData = async () => {
+    const { chainNamespace } = fromAssetId(sellAsset.assetId)
+
+    switch (chainNamespace) {
+      case CHAIN_NAMESPACE.Solana: {
+        const sellAdapter = deps.assertGetSolanaChainAdapter(sellAsset.chainId)
+        const getFeeDataInput: GetFeeDataInput<KnownChainIds.SolanaMainnet> = {
+          // Simulates a self-send, since we don't know the to just yet at this stage
+          to: input.sendAddress!,
+          value: sellAmount,
+          chainSpecific: {
+            from: input.sendAddress!,
+            tokenId:
+              sellAsset.assetId === solAssetId
+                ? undefined
+                : fromAssetId(sellAsset.assetId).assetReference,
+          },
+        }
+        const { fast } = await sellAdapter.getFeeData(getFeeDataInput)
+        return { networkFeeCryptoBaseUnit: fast.txFee }
+      }
+
+      default:
+        throw new Error('Unsupported chainNamespace')
+    }
+  }
+
+  const protocolFees: Record<AssetId, ProtocolFee> = quoteResponse.routePlan.reduce(
+    (acc, route) => {
+      const feeAssetId = toAssetId({
+        assetReference: route.swapInfo.feeMint,
+        assetNamespace: ASSET_NAMESPACE.splToken,
+        chainNamespace: CHAIN_NAMESPACE.Solana,
+        chainReference: CHAIN_REFERENCE.SolanaMainnet,
+      })
+      const feeAsset = assetsById[feeAssetId]
+
+      // If we can't find the feeAsset, we can't provide a protocol fee to display
+      // But these fees exists at protocol level, it's mostly to make TS happy as we should have the market data and assets
+      if (!feeAsset) return acc
+
+      acc[feeAssetId] = {
+        requiresBalance: false,
+        amountCryptoBaseUnit: bnOrZero(route.swapInfo.feeAmount).toString(),
```

Comment:
As a paranoia to not end up with a Frankenstein base unit amount that's actually a float
```suggestion
        amountCryptoBaseUnit: bnOrZero(route.swapInfo.feeAmount).toFixed(0),
```

Reply:
Changed in [4e76e81](https://github.com/shapeshift/web/pull/8120/commits/4e76e8115223d91bd602f7d9ceed80c94c77ea15)